logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -95,4 +99,13 @@ public FileLock obtainLock(String lockDir, String lockName) throws IOException {
             throw new LockException("Lock held by this virtual machine: " + realLockPath);
         }
     }
+
+    public void releaseLock(FileLock lock) throws IOException {
```

Comment:
only one thread has a reference to the lock at a time, right? so there is no chance that this gets called from two threads with the same lock?